### PR TITLE
Endpoint for sending a message

### PR DESCRIPTION
**Pull request by henzyd**

### CODE REVIEW

**Comment by henzyd** [comment]
METHOD: GET

Linear Link - https://linear.app/team-socket/issue/BAC-48/endpoint-for-uploading-a-file

This is a send message endpoint with a file upload parameter added to it.

Path = "/api/v1/org/{{org_id}}/rooms/{{room_id}}/messages"

This is the type:
org_id: str,
room_id: str,

Request Body =
{
sender_id: str = '619ba4671a5f54782939d385'
emojis: List[Emoji] = []
richUiData: Any = {}
files: List[AnyHttpUrl] = []
saved_by: List[str] = []
timestamp: int = 1640204440922
created_at: str = str(datetime.utcnow())
}

Response =
{
"status": "success",
"message": "new message sent",
"data": {
"sender_id": "619bab3b1a5f54782939d400",
"emojis": [],
"richUiData": {
"blocks": [
{
"key": "eljik",
"text": "Larry Gaaga",
"type": "unstyled",
"depth": 0,
"inlineStyleRanges": [],
"entityRanges": [],
"data": {}
}
],
"entityMap": {}
},
"files": [],
"saved_by": [],
"timestamp": 0,
"created_at": "2022-02-01 19:20:55.891264",
"room_id": "61e6855e65934b58b8e5d1df",
"org_id": "619ba4671a5f54782939d384",
"message_id": "61f98d0665934b58b8e5d286",
"edited": false,
"threads": []
}
}

File: txt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+anyio-3.6.2 certifi-2022.9.24 h11-0.14.0 httpcore-0.16.1 httpx-0.23.1 idna-3.4 sniffio-1.3.0
+
+fastapi-0.87.0 pydantic-1.10.2 starlette-0.21.0 typing-extensions-4.4.0
+
+installed click-8.1.3 colorama-0.4.6 uvicorn-0.20.0
+
+charset-normalizer-2.1.1 requests-2.28.1 urllib3-1.26.13
+
```
Comment (mikengr) [inline txt]: what's this for?

File: backend/schema/message.py [hunk 2]
```diff
@@ -98,9 +117,9 @@ class Thread(MessageRequest):
     """
 
     room_id: str
-    org_id: str
+    org_id: str 
     message_id: str = Field(None, alias="_id")
-    edited: bool = False
+    # message_id: str = '_id'
```
Comment (mikengr) [inline backend/schema/message.py]: Attach screenshots that shows that this works

**Review by mikengr — COMMENTED** [review]
@henzyd I left some comments.
Also what's the file `test.png` for?
Your branch has unresolved conflicts too. Always check this when you make PRs

**Comment by FlorenceEgwu** [comment]
I have a problem with the title and description of your PR. Gives no idea of what you worked on.

Your [commit messages](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/) too. 
I shouldn't need to look at your code to figure what problem you are trying to solve.

File: backend/main.py [hunk 3]
```diff
@@ -35,8 +35,8 @@
...
-app.mount(
-    "/",
-    StaticFiles(directory="../frontend/dist", html=True, check_dir=False),
-    name="static",
-)
+# app.mount(
+#     "/",
+#     StaticFiles(directory="../frontend/dist", html=True, check_dir=False),
+#     name="static",
+# )
```
Comment (FlorenceEgwu) [inline backend/main.py]: WHAT IS GOING ON HERE?
Why would you want to comment this out?

kindly take this out completely and let the file be as it was.
If you have any changes to make here, it must be properly discussed first

File: backend/schema/message.py [hunk 4]
```diff
@@ -98,9 +117,9 @@ class Thread(MessageRequest):
     """
 
     room_id: str
-    org_id: str
+    org_id: str 
     message_id: str = Field(None, alias="_id")
-    edited: bool = False
```
Comment (FlorenceEgwu) [inline backend/schema/message.py:102]: why did you remove this?

File: backend/schema/message.py [hunk 5]
```diff
@@ -98,9 +117,9 @@ class Thread(MessageRequest):
...
+    # message_id: str = '_id'
```
Comment (FlorenceEgwu) [inline backend/schema/message.py]: what is this for?
I believe line 121 above is sufficient.

**Comment by FlorenceEgwu** [comment]
You have made a lot of changes and I am doubtful they all work as expected.
kindly make a live screen recording. demonstrating the use of this functionality you have added

**Review by FlorenceEgwu — DISMISSED** [review]

File: txt (hunk 1 above)
Comment (henzyd) [reply txt]: Some dependencies where outdated I had to update them so that ours will work
I didn't want to change the requirement.txt because your code might need those versions
so I put it here I could remove it if you want but if I do you won't be able to test my code because you won't know what version I'm using

**Comment by henzyd** [comment]
image

**Comment by henzyd** [comment]
I couldn't do a screen record the file is too large 
I hope this one will suffice
@AI-fae @mikengr

**Comment by henzyd** [comment]
How do I communicate with the zc_core api 
I want to use the currently logged in user in my endpoint

I could remove it If you don't want me to add it
but the main thing is that you can add a file to a message which is what I have done

**Comment by henzyd** [comment]
The Emoji and Files parameter is required I can make it optional if you want but then It means I'll have to tamper with someone else's code

**Review by mikengr — APPROVED** [review]
LGTM

**Comment by mikengr** [comment]
Fire in the hole, hopefully nothing breaks...
